src/overlays/actors/ovl_Obj_Spinyroll/z_obj_spinyroll.c

```diff
@@ -28,121 +40,661 @@ const ActorInit Obj_Spinyroll_InitVars = {
     },
 };
 
-// static ColliderTrisInit sTrisInit = {
-static ColliderTrisInit D_80A1F1A8 = {
-    { COLTYPE_METAL, AT_ON | AT_TYPE_ENEMY, AC_ON | AC_HARD | AC_TYPE_PLAYER, OC1_NONE, OC2_NONE, COLSHAPE_TRIS, },
-    6, D_80A1F040, // sTrisElementsInit,
-};
-
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80A1F204[] = {
-    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
-    ICHAIN_F32(uncullZoneDownward, 500, ICHAIN_STOP),
+static ColliderTrisInit sTrisInit = {
+    {
+        COLTYPE_METAL,
+        AT_ON | AT_TYPE_ENEMY,
+        AC_ON | AC_HARD | AC_TYPE_PLAYER,
+        OC1_NONE,
+        OC2_NONE,
+        COLSHAPE_TRIS,
+    },
+    6,
+    sTrisElementsInit,
 };
 
-#endif
-
-extern ColliderTrisElementInit D_80A1F040[6];
-extern ColliderTrisInit D_80A1F1A8;
-extern InitChainEntry D_80A1F204[];
-
-extern UNK_TYPE D_06000460;
-extern UNK_TYPE D_06000E68;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Spinyroll/func_80A1DA50.s")
+f32 D_80A1F1B8[] = { 1.0f, 2.0f, 4.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Spinyroll/func_80A1DAAC.s")
+f32 D_80A1F1C4[] = { 1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f, 7.0f, 8.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Spinyroll/func_80A1DB2C.s")
+s16 D_80A1F1E4[] = { 1, 10, 20, 30, 40, 50, 60, 70 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Spinyroll/func_80A1DC5C.s")
+f32 D_80A1F1F4[] = { 30.0f, -30.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Spinyroll/func_80A1DCCC.s")
+f32 D_80A1F1FC[] = { 9.0f, -9.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Spinyroll/func_80A1DD18.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Spinyroll/func_80A1DE10.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Spinyroll/func_80A1DE58.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Spinyroll/func_80A1DE84.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Spinyroll/func_80A1DEB8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Spinyroll/func_80A1DFA0.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 500, ICHAIN_STOP),
+};
 
+s16 D_80A1F20C[] = { 0xFA0, -0xFA0 };
+
+void func_80A1DA50(GlobalContext* globalCtx, ObjSpinyroll* this, Vec3f* arg2, Vec3f* arg3) {
+    Vec3f sp1C;
+
+    Math_Vec3f_Sum(arg2, arg3, &sp1C);
+    Math_Vec3f_Scale(&sp1C, 0.5f);
+    EffectSsHitMark_SpawnFixedScale(globalCtx, 3, &sp1C);
+    Actor_PlaySfxAtPos(&this->dyna.actor, NA_SE_IT_SHIELD_REFLECT_SW);
+}
+
+void func_80A1DAAC(Vec3f* arg0, Vec3f* arg1, s16 arg2) {
+    f32 sp1C = Math_SinS(arg2);
+    f32 sp18 = Math_CosS(arg2);
+
+    arg0->x = (arg1->z * sp1C) + (arg1->x * sp18);
+    arg0->y = arg1->y;
+    arg0->z = (arg1->z * sp18) - (arg1->x * sp1C);
+}
+
+void func_80A1DB2C(Actor* thisx) {
+    ObjSpinyroll* this = THIS;
+    s32 i, j;
```

Comment:
```suggestion
    s32 i;
    s32 j;
```
With our styling, we don't do multiple single-line declarations